hoomd/custom_action.py

```diff
@@ -0,0 +1,114 @@
+from abc import ABC, abstractmethod
+from hoomd.parameterdicts import ParameterDict
+from hoomd.operation import _HOOMDGetSetAttrBase
+
+
+class CustomAction(ABC):
+    """Base class for all Python Action's.
+
+    This class must be the parent class for all Python ``Action``s. This class
+    requires all subclasses to implement the act method which performs the
+    Python object's task whether that be updating the system, writing output, or
+    analyzing some property of the system.
+
+    To use subclasses of this class, the object must be passed as an argument
+    to a `hoomd.update.CustomUpdater` or `hoomd.analyze.CustomAnalyzer`
+    constructor.
+
+    If the pressure, rotational kinetic energy, or external field virial is
+    needed for a subclass, the flags attribute of the class needs to be set with
+    the appropriate flags from `hoomd.util.ParticleDataFlags`.
+
+    .. code-block:: python
+
+        from hoomd.python_action import CustomAction
+        from hoomd.util import ParticleDataFlags
+
+
+        class ExampleActionWithFlag(CustomAction):
+            flags = [ParticleDataFlags.ROTATIONAL_KINETIC_ENERGY,
+                     ParticleDataFlags.PRESSURE_TENSOR,
+                     ParticleDataFlags.EXTERNAL_FIELD_VIRIAL]
+
+            def act(self, timestep):
+                pass
```

Comment:
Include in this example, or an additional example, an `act` method that actually does something (i.e. compute the center of mass of the system).